README.md

```diff
@@ -1,164 +1,190 @@
+
+The recommended Google satellite base layer can be integrated using this Leaflet plugin: https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant
+
+Here's a screenshot:
+
+![screenshot](example.png)
+
+## Demo
+
+Check out this [simple demo](https://publiclab.github.io/Leaflet.DistortableImage/examples/index.html).
+
+And watch this GIF demo:
+
+![demo gif](https://raw.githubusercontent.com/publiclab/mapknitter/master/public/demo.gif)
+
+To test the code, open `index.html` in your browser and click and drag the markers on the edges of the image. The image will show perspectival distortions.
+
+## Usage
+
+```js
+// basic Leaflet map setup
+map = new L.map('map').setView([51.505, -0.09], 13);
+L.tileLayer('https://{s}.tiles.mapbox.com/v3/anishshah101.ipm9j6em/{z}/{x}/{y}.png', {
+  maxZoom: 18,
+  attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, ' +
+    '<a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
+    'Imagery © <a href="http://mapbox.com">Mapbox</a>',
+  id: 'examples.map-i86knfo3'
+}).addTo(map);
+
+// create an image
+img = new L.DistortableImageOverlay(
+  'example.png', {
+    corners: [
+      new L.latLng(51.52,-0.10),
+      new L.latLng(51.52,-0.14),
+      new L.latLng(51.50,-0.10),
+      new L.latLng(51.50,-0.14)
+    ],
+    fullResolutionSrc: 'large.jpg', // optionally pass in a higher resolution image to use in full-res exporting
+    suppressToolbar: false // defaults to false, but you can turn off the toolbar interface and build your own
+  }
+).addTo(map);
+
+L.DomEvent.on(img._image, 'load', img.editing.enable, img.editing); // enable editing
+
+```
+
+## Full-resolution download
+
+We've added a GPU-accelerated means to generate a full resolution version of the distorted image; it requires two additional dependencies to enable; see how we've included them in the demo:
+
+
+```
+<script src="../node_modules/webgl-distort/dist/webgl-distort.js"></script>
+<script src="../node_modules/glfx-js/dist/glfx.js"></script>
+```
+## "Reveal" functionality
+
+The "reveal" tool extends the native `EditOverlayAction` Leaflet class behaviour and hides some user-specified tools, hence saving up space in the toolbar and giving more room to your mapview! Refer the comments below for a detailed explaination on what happens when the "reveal" tool is executed.
+
+```js
+function() {
+
+  /* specify an array of tools that will initially be
+   hidden inside the reveal tool in the toolbar
+   also, input is taken from the exposed API in the HTML */
+  var hidden_tools = this._overlay.options.hidden_tools, i;
+
+  /* On reveal, add all the "hidden" tools to
+  the toolbar */
+  for (i = 0; i < hidden_tools.length; i++) {
+    this._overlay._addTool(hidden_tools[i]);
+  }
+
+  /* Remove "reveal" tool from the toolbar */
+  this._overlay._removeTool(ToggleReveal);
+
+  this._overlay.editing._hideToolbar();
+}
+```
+## Adding Custom Toolbars

```

Comment:
This one is good!! Perhaps we should list the available overlays somewhere? Same clarifying concept as I mentioned above. Also maybe explain the `position` key a bit.